### PR TITLE
Update/deprecate jetpack plan tiled gallery

### PR DESCRIPTION
**Pull request by gmjuhasz**
#### Changes proposed in this Pull Request:

* Changed `Jetpack_Plans` to `Current_Plan` for tiled-gallery extenstion

Other changes connected to this:
https://github.com/Automattic/jetpack/pull/25701
https://github.com/Automattic/jetpack/pull/25706

#### Jetpack product discussion

1202726932197503-as-1202726932197515
pdWQjU-4i-p2

#### Does this pull request change what data or activity we track or use?

No.

#### Testing instructions:

* Tiled-gallery should work as before
* Try adding and using tiled gallery in editor with and without having a plan

### CODE REVIEW

**Comment by spsiddarthan** [comment]
Hi @jeherve! I needed a confidence check on testing extension files. 

Is there anything special we need to keep in mind when testing extension files such as this one `projects/plugins/jetpack/extensions/blocks/tiled-gallery/tiled-gallery.php` or will the usual jetpack watch plugins/jetpack suffice? There seem to be some special instructions [here](https://github.com/Automattic/jetpack/tree/trunk/projects/plugins/jetpack/extensions)  for the block editor extensions, which I think I should be following.

File: projects/plugins/jetpack/extensions/blocks/tiled-gallery/tiled-gallery.php [hunk 1]
```diff
@@ -61,9 +61,9 @@ public static function render( $attr, $content ) {
 
 		$is_squareish_layout = self::is_squareish_layout( $attr );
 
-		// Jetpack_Plan::get does not exist on WordPress.com.
-		if ( method_exists( 'Jetpack_Plan', 'get' ) ) {
-			$jetpack_plan = Jetpack_Plan::get();
+		// Current_Plan::get does not exist on WordPress.com.
```
Comment (jeherve) [inline projects/plugins/jetpack/extensions/blocks/tiled-gallery/tiled-gallery.php:64]: The comment won't make much sense anymore once this is merged and the package gets loaded on WordPress.com with the rest of the Jetpack plugin.

**Comment by jeherve** [comment]
> will the usual jetpack watch plugins/jetpack suffice?

It should work just fine, yeah. It will build the plugin and the editor extensions included in the plugin, so you will see them in the block editor on your site.

**Comment by gmjuhasz** [comment]
Closed this because of the age of the PR and out of scope.

**Pull request closed without merging**